analysis/dataset_definition_curation.py

```diff
@@ -113,3 +143,22 @@
             selected_events.snomedct_code.is_in(code)
         ).count_for_patient()
         dataset.add_column(f"{time_interval_desc}_count_{code_desc}", count_codes_query)
+
+# Get all medications over 1 year
+selected_medications_any = (
+    medications.where(
+        medications.date.is_on_or_between(
+            "2023-05-31",
+            "2024-04-30",
+        )
+    )
+)
+
+# Count all medication statuses
+for key, desc in med_status_dict.items():
```

Comment:
let's iterate through integers from 0 to 28 instaed of the dict here, we can also use the number of the medication status for the column name, no need to have a more descriptive string in the new column name at this stage